design-documents/storefront/pricing.md

```diff
@@ -0,0 +1,355 @@
+}
+
+message PriceBookUnassignProductsResult {
+    int32 status = 1;
+}
+
+
+message GetPricesInput {
+    string price_book_id = 1;
+    repeated string product_ids = 2;
+}
+
+message GetPricesOutput {
+    message ProductPrice {
+        string product_id = 2;
+        // Price without applied discounts
+        float regular_price = 3;
+        // Price with applied discounts
+        float price = 4;
+        float minimum_price = 5;
+        float maximum_price = 6;
+    }
+
+    repeated ProductPrice prices = 1;
+}
+
+
+service PriceBook {
+    rpc create(PriceBookInput) returns (PriceBookCreateResult);
+    rpc delete(PriceBookDeleteInput) returns (PriceBookDeleteResult);
+    rpc assignProducts(PriceBookInput) returns (PriceBookAssignProductsResult);
+    rpc unassignProducts(UnassignProducts) returns (PriceBookUnassignProductsResult);
+    rpc getPrices(GetPricesInput) returns (GetPricesOutput);
+}
+```
+
+### Customer tags instead of customer groups
+
+Magento monolith uses `customer groups` for customer segmentation globally. There is one-to-many relation between customer groups and customers, 
+so customer could be a member of excatly one group only. However, in modern world, each customer could be a 
+member of different groups based on current behavior. For example, pricing system works with wholesale and regular buyers,
+but recommendation system works with different groups of customers which are based on gender, age, ML-generated groups, etc.
+
+In order to provide more flexibility in customer segmentation, we may introduce many-to-many. Also, having `customer groups` 
+which are not bound to pricing functionality make them looks like a regular tags. Thus, we may also rename them to tags:
+
+![Customer tags](pricing/customer-tags.png)
+
+### Complex products support
+
+The `minimum prices` of complex products calculated based on variation's prices, variation's availability and variation's stock.
+Having variations as a separate products makes `minimum price` and `maximum price` dependent on products which may not
+be visible for the current group or in a current catalog. Example: configurable product contains variation #1 - price $10,
+ 2 - price $9 and 3 - price $12. Let's imagine that variation #2 is visible for people with "VIP access" only,
+then the desired `minimum price` of configurable product for basic access will be $10, for "VIP access" - $9. However, there is 
+only one price book in the system, so we can hold only one value.
+
+This happens because parent product and variation are separate products which could be assigned to different access lists 
+and price books. In order to mitigate this issue products should be isolated, so product options fully define complex products.
+
+The case from example above could be handled by two independent configurable products with different sets of variations 
+for different access lists.
+
+Details will be provided in the separate proposal.
+
+### Prices fallback
```

Comment:
How many fallbacks will be allowed to set up for the default price book? Is it possible to set up a fallback for the fallback price book?